### PR TITLE
Blog App - Add Views

### PR DESCRIPTION
**Pull request by rmiyoyo**
## Description
In this project I added the Views associated with the Models and Controllers I already created for my Blog App.

### Project requirements
- Implemented the design from the sneak peek wireframes. I did not add style as it is not compulsory at the moment.
- Analyzed the wireframes deeply and identified all the necessary elements and how a user would navigate from one page to another.    
- Used methods that I have created in the Blog App - processing data in models.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+
+        <div id="bio" class="border-4 border-black mt-4 md:w-1/2 mx-auto mb-5">
+          <h2>Bio</h2>
+          <p><%= @user.bio %></p>
+        </div>
+  </div>
+
+  <% @user.posts.each do |post| %>
+    <%= render partial: 'shared/post', locals: {post: post} %>
+  <% end %>
```
Comment (tugsem) [inline app/views/users/show.html.erb:13]: - [ ] As per the requirement, please call the method here as you created in your user model to call recent three posts.

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-  <body>
-    <h1>Posts for a given User</h1>
-  </body>
-</html>
+<div>
+  <%= render partial: 'shared/user', locals: {user: @user} %>
+
+  <% @posts.each do |post| %>
+    <%= render partial: 'shared/post', locals: {post: post} %>
+    <%= render partial: 'shared/comment', locals: {post: post} %>
```
Comment (tugsem) [inline app/views/posts/index.html.erb:6]: - [ ] Here as well, please use the method that you created in your post model which returns last 5 comments.

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,9 +1,5 @@
...
-  <body>
-    <h1>A list of all users</h1>
-  </body>
-</html>
\ No newline at end of file
+<div class="flex flex-col w-full gap-y-5">
+  <% @users.each do |user| %>
+    <%= render partial: 'shared/user', locals: {user: user} %>
+  <% end %>
+</div>
```
Comment (tugsem) [inline app/views/users/index.html.erb:5]: - [ ] Please fix the layout for the user to match the given wireframe. Which you can arrange with flexbox. User photo, username should be on the same line watching the space between them. The post counter should be in the right bottom.
image
image

File: app/views/shared/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<div id="post" class="border-solid border-4 border-black md:w-1/2 mx-auto w-full mb-5">
+  <h2 class="font-bold ml-3"><%= post.title %></h2>
+  <p class="mt-5 ml-3" ><%= post.text %></p>
+  <p class="mt-5 right-2 ml-3">comments: <span class="font-bold"><%= post.comments_counter%>, </span>Likes: <span class="font-bold"><%= post.likes_counter %></span></p>
```
Comment (tugsem) [inline app/views/shared/_post.html.erb:4]: - [ ] Please give some space around the components as in the given wireframe, also the counters should be on the right bottom.
image
image

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1,9 +1,19 @@
...
+<div class="flex-col">
+  <div id="post" class="border-solid border-4 border-black md:w-1/2 mx-auto w-full mb-5">
+    <div class="flex justify-between">
+      <div>
+        <h2 class="font-bold ml-3"><%= @post.title %><span class="font-normal"> by <%= @user.name %></span></h2>
+      </div>
+
+      <div class="mr-5">
+        <p>comments: <span class="font-bold"><%= @post.comments_counter%>, </span>Likes: <span class="font-bold"><%= @post.likes_counter %></span></p>
+      </div>
```
Comment (tugsem) [inline app/views/posts/show.html.erb:10]: - [ ] Please move this counter section to the right side to match the given design.
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @rmiyoyo  👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️linters are passing
✔️correct use of gitflow
✔️views for user and post created correctly
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
## Description
In this project I added the Views associated with the Models and Controllers I already created for my Blog App.

### Project requirements
- Implemented the design from the sneak peek wireframes. I did not add style as it is not compulsory at the moment.
- Analyzed the wireframes deeply and identified all the necessary elements and how a user would navigate from one page to another.    
- Used methods that I have created in the Blog App - processing data in models.

**Review by julie-ify — COMMENTED** [review]
Hi @rmiyoyo,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly submit a new PR without reviewer comments as this one, from last week, was not approved.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**